packages/synapse-react-client/src/lib/utils/hooks/SynapseAPI/file/useFiles.ts

```diff
@@ -70,3 +74,34 @@ export function useGetProfileImage(
     },
   )
 }
+
+/**
+ * Get a batch of files from the backend.
+ * This hook not support fetching pre-signed URLs. In most web UI cases, you will want to use
+ * {@link SynapseClient#getPortalFileHandleServletUrl}
+ * @param request
+ * @param options
+ */
+export function useGetFileBatch(
+  request: BatchFileRequest,
+  options?: UseQueryOptions<BatchFileResult, SynapseClientError>,
+) {
+  const { accessToken, keyFactory } = useSynapseContext()
+  const queryFn = async () => SynapseClient.getFiles(request, accessToken)
+
+  if (request.includePreSignedURLs || request.includePreviewPreSignedURLs) {
+    // Don't use this hook if you need pre-signed URLs. They expire every 30 seconds, so you will either end up giving the
+    // user an expired URL, or making requests to the backend more frequently than necessary.
+    console.error('useGetFileBatch does not support pre-signed URLs.')
+    request.includePreSignedURLs = false
+    request.includePreviewPreSignedURLs = false
+  }
```

Comment:
It would be totally inappropriate to use this hook to get a pre-signed URL, since we'd have to get a new one every 30s. Considered throwing an error but decided to just alter the request and log an error.

Reply:
My preference would be to throw an error to fail in a highly visible way rather than update the caller's request to a valid value, but if the caller is trying to get a batch of presigned URLs using this hook then they should eventually find this code